include/r_index_f/r_index_f.hpp

```diff
@@ -413,16 +409,37 @@ class r_index_f
 
     /*
      * \param Run position (RLBWT)
-     * \param Current character offset in block
+     * \param Current offset within interval
+     * \param Character to step from
      * \return run position and offset of preceding character
      */
-    std::pair<ulint, ulint> LF(ulint run, ulint offset)
+    std::pair<ulint, ulint> LF(ulint run, ulint offset, char c)
     {
-        i_block* curr = &B_table[run/block_size];
-        const ulint k = run%block_size;
+        ulint b = run/block_size;
+        ulint k = run%block_size;
+        i_block* curr = &B_table[b];
+
+        auto [c_rank, bwt_c] = curr->heads.inverse_select(k);
+        if (c != bwt_c)
```

Comment:
At worst, it didn't break inversion. Will test more when computing count queries